### PR TITLE
Update rate limit documentation

### PR DESCRIPTION
**Pull request by nbarbettini**
## Description:
- Update rate limits for Enterprise orgs, and add limits for Developer (free), Developer (paid), and One App orgs.
- Some limits are now the same, and don't need to be printed separately. Those rows have been removed.
- Clean up minor bugs in text.

### Resolves:
[OKTA-169183](https://oktainc.atlassian.net/browse/OKTA-169183)

### CODE REVIEW

**Comment by nbarbettini** [comment]
The rate limit documentation has been updated with the latest limits.

@prasanthmathialagan-okta @jaikumarbathija-okta @simranbrar-okta Please review this and verify that the information is correct. Thanks!

Here's a downloadable version as a quick preview: [Rate Limiting at Okta _ Okta Developer.pdf](https://github.com/okta/okta.github.io/files/2019050/Rate.Limiting.at.Okta._.Okta.Developer.pdf)

File: _source/_docs/api/getting_started/rate-limits.md [hunk 1]
```diff
@@ -22,65 +22,73 @@ Rate limits may be changed to protect customers. We provide advance warning of c
...
+Note that limits for more specific endpoints override the limits for less specific endpoints. For example, the limit for getting an application by ID (second row) is higher than the more general limit for the `/api/v1/apps` endpoint (first row).
+
+| Action and Okta API Endpoint | Developer (free) | Developer (paid) & One App | Enterprise |
+|----------------------------- | ---------------- | -------------------------- | ---------- |
+| **Creating or listing applications:**<br>`/api/v1/apps` except `/api/v1/apps/{id}`        | 20 | 25 |  50 |
+| **Get, update, or delete an application by ID:**<br>`/api/v1/apps/{id}` only              | 100 | 300 | 600 |
+| **Get System Log data:**<br>`/api/v1/logs`                                                | 20 | 25 | 50 |
+| **Get System Log data:**<br>`/api/v1/events`                                              | 20 | 25 | 50 |
+| **Creating, updating, or deleting users:**<br>`/api/v1/users` except `/api/v1/users/{login}`  | 100  | 300 | 600 |
+| **Get a user by login:**<br>`/api/v1/users/{login}` only                                  | 100 | 300 | 1000 |
```
Comment (nbarbettini) [inline _source/_docs/api/getting_started/rate-limits.md]: Is 1000 for Enterprise correct here?

File: _source/_docs/api/getting_started/rate-limits.md [hunk 2]
```diff
@@ -22,65 +22,73 @@ Rate limits may be changed to protect customers. We provide advance warning of c
...
 ### Okta Rate Limits for All Other Endpoints
-Finally, for all endpoints not listed in the tables above, the API rate limit is a combined 10,000 requests per minute. 
+
+Finally, for all endpoints not listed in the tables above, the API rate limit is a combined rate limit:
+
+| Developer (free) | Developer (paid) & One App    | Enterprise |
+| ---------------- | ----------------------------- | ---------- |
+| 1000             | 3,000                         | 6,000      |
+
+For organizations created before 2018-05-22, and all IT Products orgs, the limit is 10,000 requests per minute.
```
Comment (nbarbettini) [inline _source/_docs/api/getting_started/rate-limits.md]: Is the statement "and all IT Products orgs" correct?

File: _source/_docs/api/getting_started/rate-limits.md (hunk 1 above)
Comment (prasanthmathialagan-okta) [reply _source/_docs/api/getting_started/rate-limits.md]: This must be 
Get a user by user ID or login (combined) | `/api/v1/users/{id}` or `/api/v1/users/{login}`  only

File: _source/_docs/api/getting_started/rate-limits.md [hunk 3]
```diff
@@ -22,65 +22,73 @@ Rate limits may be changed to protect customers. We provide advance warning of c
...
-| All other actions | `/api/v1/`                                               |  1200 |
+Note that limits for more specific endpoints override the limits for less specific endpoints. For example, the limit for getting an application by ID (second row) is higher than the more general limit for the `/api/v1/apps` endpoint (first row).
+
+| Action and Okta API Endpoint | Developer (free) | Developer (paid) & One App | Enterprise |
+|----------------------------- | ---------------- | -------------------------- | ---------- |
+| **Creating or listing applications:**<br>`/api/v1/apps` except `/api/v1/apps/{id}`        | 20 | 25 |  50 |
+| **Get, update, or delete an application by ID:**<br>`/api/v1/apps/{id}` only              | 100 | 300 | 600 |
+| **Get System Log data:**<br>`/api/v1/logs`                                                | 20 | 25 | 50 |
+| **Get System Log data:**<br>`/api/v1/events`                                              | 20 | 25 | 50 |
+| **Creating, updating, or deleting users:**<br>`/api/v1/users` except `/api/v1/users/{login}`  | 100  | 300 | 600 |
```
Comment (prasanthmathialagan-okta) [inline _source/_docs/api/getting_started/rate-limits.md]: `/api/v1/users` is used only for creating a user or listing users. `/api/v1/users/{idOrLogin}` will be used for deleting or updating a user

File: _source/_docs/api/getting_started/rate-limits.md [hunk 4]
```diff
@@ -22,65 +22,73 @@ Rate limits may be changed to protect customers. We provide advance warning of c
...
+| **Get a user by login:**<br>`/api/v1/users/{login}` only                                  | 100 | 300 | 1000 |
+| **Create an organization:**<br>`/api/v1/orgs`                                             | N/A | N/A | 50 |
+| **Authentication with a custom Authorization Server:**<br>`/oauth2/{authServerId}/v1/authorize` and `/oauth2/{authServerId}/v1/token` | 100 | 300 | 600 |
```
Comment (prasanthmathialagan-okta) [inline _source/_docs/api/getting_started/rate-limits.md]: These two end points have separate limits of 300 and 600 not combined

File: _source/_docs/api/getting_started/rate-limits.md (hunk 1 above)
Comment (prasanthmathialagan-okta) [reply _source/_docs/api/getting_started/rate-limits.md]: 1000 is correct

File: _source/_docs/api/getting_started/rate-limits.md (hunk 3 above)
Comment (prasanthmathialagan-okta) [reply _source/_docs/api/getting_started/rate-limits.md]: There must be a separate entry here which says,

Updating or deleting a user by ID with 100,300,600

File: _source/_docs/api/getting_started/rate-limits.md (hunk 1 above)
Comment (nbarbettini) [reply _source/_docs/api/getting_started/rate-limits.md]: @prasanthmathialagan-okta According to the internal wiki, `/api/v1/users/{id}` is 100/300/600, while `/api/v1/users/{login}` is 100/300/1000. That's why I only called out "Get a user by login". Am I misunderstanding something?

File: _source/_docs/api/getting_started/rate-limits.md (hunk 2 above)
Comment (nbarbettini) [reply _source/_docs/api/getting_started/rate-limits.md]: Removed

File: _source/_docs/api/getting_started/rate-limits.md [hunk 5]
```diff
@@ -315,18 +315,18 @@ Extensions to the base URLs listed below are included in the specified limit, un
 
 | Action and Okta API Endpoint                                                                           | Legacy Enterprise |
 | ------------------------------------------------------------------------------------------------------ | :---------------: |
-| **Create or list apps:**<br>`/api/v1/apps` except `/api/v1/apps/{id}`                                  | 100               |
-| **Get, update, or delete an application:**<br>`/api/v1/apps/{id}`                                      | 500               |
 | **Authenticate different end users:**<br>`/api/v1/authn`                                               | 500               |
 | **Verify factors:**<br>`/api/v1/authn/factors/{ID}/verify` only                                        | No limit          |
```
Comment (prasanthmathialagan-okta) [inline _source/_docs/api/getting_started/rate-limits.md]: It falls under /api/v1/authn itself. No limit is kind of misleading. I would remove this if I were you

File: _source/_docs/api/getting_started/rate-limits.md [hunk 6]
```diff
@@ -315,18 +315,18 @@ Extensions to the base URLs listed below are included in the specified limit, un
...
-| **Get session information:**<br>`/api/v1/sessions`                                                     | 750               |
 | **Create or list users:**<br>`/api/v1/users` except `/api/v1/users/{id}` and `/api/v1/users/{login}`   | 600               |
 | **Get a user by user ID or login (combined):**<br>`/api/v1/users/{id}` or `/api/v1/users/{login}` only | 2000              |
 | **Update or delete a user by ID:**<br>`/api/v1/users/{id}` only                                        | 600               |
+| **Get System Log data:**<br>`/api/v1/logs`                                                             | 120               |
+| **Get System Log data:**<br>`/api/v1/events`                                                           | No limit          |
+| **Get session information:**<br>`/api/v1/sessions`                                                     | 750               |
 | **Create an org (ISVs only):**<br>`/api/v1/orgs` (not available in One App)                            | 50                |
 | **Authentication with Custom Authorization Servers:**<br>`/oauth2/{authServerId}/v1/authorize`         | No limit          |
```
Comment (prasanthmathialagan-okta) [inline _source/_docs/api/getting_started/rate-limits.md]: Can we just remove the entries which says "No limit"?

**Comment by nbarbettini** [comment]
> It is better to mention, for legacy orgs, concurrency limit is still 75

Done, thanks @prasanthmathialagan-okta

**Pull request merged**